core/src/main/scala/ai/lum/odinson/serialization/JsonSerializer.scala

```diff
@@ -348,4 +378,32 @@ object JsonSerializer {
     }
   }
 
+  // -------------------------------------
+  //        HELPER METHODS
+  // -------------------------------------
+
+  def checkVerbosity(verbose: String): Unit = {
+    if (!validVerbose.contains(verbose)) {
+      throw new OdinsonException(
+        s"Invalid setting for `verbose`: $verbose [valid: ${validVerbose.mkString(", ")}]"
+      )
+    }
+  }
+
+  def checkEngine(verbose: String, engine: Option[ExtractorEngine]): Unit = {
+    if (verbose != NONE && engine.isEmpty) {
+      throw new OdinsonException(
+        "Cannot request verbose serialization without providing an ExtractorEngine."
+      )
+    }
+  }
+
+}
+
+object JsonSerializer {
+  val NONE = "none"
```

Comment:
enums -- then the users need to pass in ENUM
`new JsonSerializer(enum.NONE)...` (pseudocode)